api/src/main/java/marquez/db/LineageDao.java

```diff
@@ -102,16 +102,27 @@ INNER JOIN jobs_view j ON (j.uuid=l2.job_uuid OR j.uuid=l2.job_symlink_target_uu
       """
       SELECT ds.*, dv.fields, dv.lifecycle_state
       FROM datasets_view ds
-      LEFT JOIN dataset_versions dv on dv.uuid = ds.current_version_uuid
-      WHERE ds.uuid IN (<dsUuids>)""")
+      LEFT JOIN dataset_versions dv ON dv.uuid = ds.current_version_uuid
+      LEFT JOIN dataset_symlinks dsym ON dsym.namespace_uuid = ds.namespace_uuid and dsym.name = ds.name
+      WHERE dsym.is_primary = true
+      AND ds.uuid IN (<dsUuids>)""")
```

Comment:
So here since the view datasets_views can have several rows with the same uuid we choose the one flagged as primary.